src/linker/rule.py

```diff
@@ -0,0 +1,86 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Callable, List
+
+
+class Rule(ABC):
+    def write_to_snakefile(self, snakefile_dir) -> None:
+        snakefile = snakefile_dir / "Snakefile"
+        with open(snakefile, "a") as f:
+            f.write(self._build_rule())
+
+    @abstractmethod
+    def _build_rule(self) -> str:
+        pass
+
+
+@dataclass
+class TargetRule(Rule):
```

Comment:
I think docstrings for these classes will be useful

Reply:
good call, added